src/Browscap/Data/Browser.php

```diff
@@ -20,19 +25,29 @@ class Browser
     private $isStandard = false;
 
     /**
-     * @param string[] $properties
-     * @param bool     $isLite
-     * @param bool     $standard
+     * @param array  $properties
+     * @param string $type
+     * @param bool   $isLite
+     * @param bool   $standard
      */
-    public function __construct(array $properties, bool $isLite, bool $standard)
+    public function __construct(array $properties, string $type, bool $isLite, bool $standard)
     {
+        $this->type       = $type;
```

Comment:
What values can `type` be?

Reply:
Do want just a list of these types, or do you want a proper validation?

Reply:
This object is a simple data container. The validation has to be setup in the factory.

Reply:
Data should be validated as it enters the place of usage (i.e. the class), rather than assume it is validated somewhere else; that's why value objects are useful; encapsulate validation of values in a value object, make it immutable so it can never accidentally become invalid (if you want to change the value, you create a new object always), and there is implicit trust that the value object is in a consistent, valid and invariant state.
Specifically in this case, it was a question really: is `$type` really an enumeration? Can the value be validated?

Reply:
I added the validation for the `type` property in the factory, like all other validation we do.

The type property is not really an enum. The possible values are limited by the external libraries I created and used here and by the validation `browscap-php` does.